diff --git a/app/src/main/java/app/olaunchercf/MainViewModel.kt b/app/src/main/java/app/olaunchercf/MainViewModel.kt
index 38749bb..ab4f0c4 100644
--- a/app/src/main/java/app/olaunchercf/MainViewModel.kt
+++ b/app/src/main/java/app/olaunchercf/MainViewModel.kt
@@ -52,6 +52,10 @@ class MainViewModel(application: Application) : AndroidViewModel(application) {
                 prefs.appSwipeRight = appModel
                 updateSwipeApps()
             }
+            AppDrawerFlag.SetSwipeDown -> {
+                prefs.appSwipeDown = appModel
+                updateSwipeApps()
+            }
             AppDrawerFlag.SetClickClock -> {
                 prefs.appClickClock = appModel
                 updateClickApps()
diff --git a/app/src/main/java/app/olaunchercf/data/Constants.kt b/app/src/main/java/app/olaunchercf/data/Constants.kt
index 688527d..5ab7890 100644
--- a/app/src/main/java/app/olaunchercf/data/Constants.kt
+++ b/app/src/main/java/app/olaunchercf/data/Constants.kt
@@ -43,6 +43,7 @@ object Constants {
         SetHomeApp,
         SetSwipeLeft,
         SetSwipeRight,
+        SetSwipeDown,
         SetClickClock,
         SetClickDate,
     }
diff --git a/app/src/main/java/app/olaunchercf/data/Prefs.kt b/app/src/main/java/app/olaunchercf/data/Prefs.kt
index 4335dbd..a54905d 100644
--- a/app/src/main/java/app/olaunchercf/data/Prefs.kt
+++ b/app/src/main/java/app/olaunchercf/data/Prefs.kt
@@ -28,6 +28,7 @@ private const val HOME_LOCKED = "HOME_LOCKED"
 private const val SHOW_TIME = "SHOW_TIME"
 private const val SWIPE_LEFT_ENABLED = "SWIPE_LEFT_ENABLED"
 private const val SWIPE_RIGHT_ENABLED = "SWIPE_RIGHT_ENABLED"
+private const val SWIPE_DOWN_ENABLED = "SWIPE_DOWN_ENABLED"
 private const val CLICK_CLOCK_ENABLED = "CLICK_CLOCK_ENABLED"
 private const val CLICK_DATE_ENABLED = "CLICK_DATE_ENABLED"
 private const val SCREEN_TIMEOUT = "SCREEN_TIMEOUT"
@@ -44,6 +45,7 @@ private const val APP_ACTIVITY = "APP_ACTIVITY"
 
 private const val SWIPE_RIGHT = "SWIPE_RIGHT"
 private const val SWIPE_LEFT = "SWIPE_LEFT"
+private const val SWIPE_DOWN = "SWIPE_DOWN"
 private const val CLICK_CLOCK = "CLICK_CLOCK"
 private const val CLICK_DATE = "CLICK_DATE"
 
@@ -146,6 +148,10 @@ class Prefs(val context: Context) {
         get() = prefs.getBoolean(SWIPE_RIGHT_ENABLED, true)
         set(value) = prefs.edit().putBoolean(SWIPE_RIGHT_ENABLED, value).apply()
 
+    var swipeDownEnabled: Boolean
+        get() = prefs.getBoolean(SWIPE_DOWN_ENABLED, true)
+        set(value) = prefs.edit().putBoolean(SWIPE_DOWN_ENABLED, value).apply()
+
     var clickClockEnabled: Boolean
         get() = prefs.getBoolean(CLICK_CLOCK_ENABLED, true)
         set(value) = prefs.edit().putBoolean(CLICK_CLOCK_ENABLED, value).apply()
@@ -207,6 +213,10 @@ class Prefs(val context: Context) {
         get() = loadApp(SWIPE_LEFT)
         set(appModel) = storeApp(SWIPE_LEFT, appModel)
 
+    var appSwipeDown: AppModel
+        get() = loadApp(SWIPE_DOWN)
+        set(appModel) = storeApp(SWIPE_DOWN, appModel)
+
     var appClickClock: AppModel
         get() = loadApp(CLICK_CLOCK)
         set(appModel) = storeApp(CLICK_CLOCK, appModel)
diff --git a/app/src/main/java/app/olaunchercf/ui/AppDrawerFragment.kt b/app/src/main/java/app/olaunchercf/ui/AppDrawerFragment.kt
index ce08304..301b545 100644
--- a/app/src/main/java/app/olaunchercf/ui/AppDrawerFragment.kt
+++ b/app/src/main/java/app/olaunchercf/ui/AppDrawerFragment.kt
@@ -55,15 +55,16 @@ class AppDrawerFragment : Fragment() {
                 binding.drawerButton.setOnClickListener { renameListener(flag, n) }
             }
             AppDrawerFlag.SetSwipeRight,
-                AppDrawerFlag.SetSwipeLeft,
-                AppDrawerFlag.SetClickClock,
-                AppDrawerFlag.SetClickDate -> {
-                    binding.drawerButton.text = getString(R.string.disable)
-                    binding.drawerButton.isVisible = true // && it.trim().isNotEmpty()
-                    binding.drawerButton.setOnClickListener {
-                        disableGesture(flag)
-                        findNavController().popBackStack()
-                    }
+            AppDrawerFlag.SetSwipeLeft,
+            AppDrawerFlag.SetSwipeDown,
+            AppDrawerFlag.SetClickClock,
+            AppDrawerFlag.SetClickDate -> {
+                binding.drawerButton.text = getString(R.string.disable)
+                binding.drawerButton.isVisible = true // && it.trim().isNotEmpty()
+                binding.drawerButton.setOnClickListener {
+                    disableGesture(flag)
+                    findNavController().popBackStack()
+                }
             }
             else -> {}
         }
@@ -218,6 +219,9 @@ class AppDrawerFragment : Fragment() {
             AppDrawerFlag.SetSwipeRight -> {
                 prefs.swipeRightEnabled = false
             }
+            AppDrawerFlag.SetSwipeDown -> {
+                prefs.swipeDownEnabled = false
+            }
             AppDrawerFlag.SetClickClock -> {
                 prefs.clickClockEnabled = false
             }
diff --git a/app/src/main/java/app/olaunchercf/ui/HomeFragment.kt b/app/src/main/java/app/olaunchercf/ui/HomeFragment.kt
index a60f4ef..7f01adc 100644
--- a/app/src/main/java/app/olaunchercf/ui/HomeFragment.kt
+++ b/app/src/main/java/app/olaunchercf/ui/HomeFragment.kt
@@ -194,6 +194,13 @@ class HomeFragment : Fragment(), View.OnClickListener, View.OnLongClickListener
         else openDialerApp(requireContext())
     }
 
+    private fun openSwipeDownApp() {
+        if (!prefs.swipeDownEnabled) return
+        if (prefs.appSwipeDown.appPackage.isNotEmpty())
+            launchApp(prefs.appSwipeDown)
+        else openDialerApp(requireContext())
+    }
+
     private fun openClickClockApp() {
         if (!prefs.clickClockEnabled) return
         if (prefs.appClickClock.appPackage.isNotEmpty())
@@ -254,7 +261,8 @@ class HomeFragment : Fragment(), View.OnClickListener, View.OnLongClickListener
 
             override fun onSwipeDown() {
                 super.onSwipeDown()
-                expandNotificationDrawer(context)
+                openSwipeDownApp()
+                // expandNotificationDrawer(context)
             }
 
             override fun onLongClick() {
@@ -309,7 +317,8 @@ class HomeFragment : Fragment(), View.OnClickListener, View.OnLongClickListener
 
             override fun onSwipeDown() {
                 super.onSwipeDown()
-                expandNotificationDrawer(context)
+                openSwipeDownApp()
+                // expandNotificationDrawer(context)
             }
 
             override fun onLongClick(view: View) {
diff --git a/app/src/main/java/app/olaunchercf/ui/SettingsFragment.kt b/app/src/main/java/app/olaunchercf/ui/SettingsFragment.kt
index 089a6f2..75438a3 100644
--- a/app/src/main/java/app/olaunchercf/ui/SettingsFragment.kt
+++ b/app/src/main/java/app/olaunchercf/ui/SettingsFragment.kt
@@ -303,6 +303,17 @@ class SettingsFragment : Fragment() {
                             active = prefs.swipeRightEnabled,
                         )
                     },
+                    { _, _ ->
+                        SettingsAppSelector(
+                            title = stringResource(R.string.swipe_down_app),
+                            currentSelection = remember {
+                                mutableStateOf(prefs.appSwipeDown.appLabel.ifEmpty { "Phone" })
+                            },
+
+                            onClick = { updateGesture(AppDrawerFlag.SetSwipeDown) },
+                            active = prefs.swipeDownEnabled,
+                        )
+                    },
                     { _, _ ->
                         SettingsAppSelector(
                             title = stringResource(R.string.clock_click_app),
